internal/handler/handlers.go

```diff
@@ -3318,6 +3321,12 @@ func RetrieveMediaPageHandler(svr server.Server) http.HandlerFunc {
 			svr.JSON(w, http.StatusInternalServerError, nil)
 			return
 		}
+
+		// We blur the image if the user is not logged in
+		if profile, _ := middleware.GetUserFromJWT(r, svr.SessionStore, svr.GetJWTSigningKey()); profile == nil {
```

Comment:
this will blur company and jobs images as well, it should only blur dev images

Reply:
How do you think this could be fixed?

I think the blurring should be when you ask for the developer’s profile, and return the id of the blurred image instead of the original one. But then we’ll have one of two choices, 
- Save the blurred images in the database and having a separate id for them in the developer profile entity, this should be much better than this approach because the the blurred images will be pre-baked and no need to blur the image with each request.
- The above solution is a little complex and its only benefit is that each blurred image will _look_ like the original one.In this solution we can simply have a general blurred image(or multiple if we want to get creative) and return it instead of the actual blurred one.

let me know what you think

Reply:
you can try add a join on the database.GetMediaByID function, this should only join developer table. you will need to update the return struct/object. then in this function if the dev id is set and if the user is not authenticated you will blur the image

Reply:
done